src/main/java/org/synyx/urlaubsverwaltung/core/account/service/AccountInteractionServiceImpl.java

```diff
@@ -35,12 +35,25 @@
     }
 
     @Override
-    public Account createHolidaysAccount(Person person, DateMidnight validFrom, DateMidnight validTo,
+    public synchronized Account createHolidaysAccount(Person person, DateMidnight validFrom, DateMidnight validTo,
```

Comment:
a test for this code is missing.

Reply:
why do we need synchronized? Are their parallel thread?

Reply:
> why do we need synchronized? Are their parallel thread?

Yes, that is the whole point of this pull request.

If the API gets called twice concurrently, multiple holiday accounts are created for the same person and year, resulting in database errors described in #443.

It would be better though, to use a unique constraint inside of the database to enforce this, as a `synchronized` block can only work within a single JVM instance and will not reliably fix the problem when you have a clustered setup (or if other code somewhere also accidentally creates duplicate entries).

Reply:
I provided a solution, maybe you want to take a look at it.